src/lib.rs

```diff
@@ -670,13 +674,17 @@ where
         loop {
             // Send stuff
             // TODO flow control
-            if let IoctlState::Pending { kind, cmd, iface, buf } = self.state.ioctl_state.get() {
-                self.send_ioctl(kind, cmd, iface, unsafe { &*buf }).await;
-                self.state.ioctl_state.set(IoctlState::Sent { buf });
-            }
+            if self.sdpcm_seq == self.tx_seq_max || self.tx_seq_max.wrapping_sub(self.sdpcm_seq) & 0x80 != 0 {
```

Comment:
It's possible this check passes but we only have window to send 1 packet, but the below code could send 2 packets (an ioctl and an ethernet packet)